lib/ukdebug/Config.uk

```diff
@@ -136,4 +136,17 @@ config LIBUKDEBUG_GDBSTUB
         this ID will be used to communicate with GDB. If there is only
         one console device on the system, its ID is always 0.
 
+if LIBUKDEBUG_GDBSTUB
+config LIBUKDEBUG_GDB_ALWAYS_ACK
+	bool "Always use acknowledgment packets"
```

Comment:
What do you think of: `Force "always ack" mode`

Should we also provide a kernel parameter option for this?

Reply:
The only reason to touch this option is when using an unreliable serial connection. Idk if we need a kernel parameter to dynamically set this option. Your choice

Reply:
I don't think "force" accurately describes what's going on. We don't force anything here. In fact, using acknowledgment packets is the default.

Reply:
I think the kernel parameter might be useful but you can keep it as is, as well (can be added later if needed). It would be interesting only for bare metal targets and we have currently a focus on virtualization environments.

It seems I did not well understand the GDB ACK option. Is the degree of ack'ing negotiated with gdb? Can also gdb ask for this and the option changes the initial default?

Reply:
By default ack packets are always sent. The gbd host and the stub negotiate the available features right after connecting. At this stage, the stub can signal to the host that the stub wants to disable ack packets. Then, the gdb host can send the stub a packet to actually switch off ack packets. But this is up to the host; the stub can't control if this packet is sent by the stub or not. If ack packets are switched off, they are off forever. This is why even if we try to use "no ack" mode, it might turn out that the gdb host never sends us the packet to turn off ack packets, and as such ack packets stay on

Reply:
Thanks for the explanation. Makes much more sense now. As discussed offline we spoke about a slight update to the help summary to the option to make it slightly clearer.